File: DesignPatternsExercises/ruby/mediator_design_pattern/solution/seller.rb
class Seller
  attr_reader :currency, :selling_price

  def initialize(name, mediator, currency, selling_price)
    @name = name
    @currency = currency
    @selling_price = selling_price
    # mediator.__________(7)__________
    mediator.add_seller(self)
  end

  # def __________(8)__________(bid_in_dollars)
  def bid_accepted?(bid_in_dollars)
    if bid_in_dollars <= selling_price
      puts "Seller #{@name} accepts the bid of USD #{bid_in_dollars}"
      @selling_price = 0
      true
    else
      puts"Seller #{@name} rejects the bid of USD #{bid_in_dollars}"
      false
    end
  end
end

File: DesignPatternsExercises/java/MediatorDesignPattern/Exercise/Seller.java
public class Seller {
	private String name;
	private String currency;
	private int sellingPrice;

	public Seller(String name, Mediator mediator, String currency, int sellingPrice) {
		this.name = name;
		this.currency = currency;
		this.sellingPrice = sellingPrice;
		mediator.__________(6)__________;
	}

	public String getCurrency() {
		return currency;
	}

	public int getSellingPrice() {
		return sellingPrice;
	}

	public boolean __________(7)__________(float bidInDollars) {
		if (bidInDollars <= sellingPrice) {
			System.out.println("Seller " + name + " accepts the bid of USD " + bidInDollars);
			sellingPrice = 0;
			return true;
		} else {
			System.out.println("Seller " + name + " rejects the bid of USD " + bidInDollars);
			return false;
		}
	}
}
